### DISCUSSION TITLE
post/android-kvm-on-mediatek/

### DISCUSSION THREAD

**Discussion by giscus** [lyc8503/lyc8503#17] [Announcements]
# post/android-kvm-on-mediatek/

在上一篇博客创作的过程中, 我意外发现我目前的主力机 Redmi Note 10 Pro 5G (chopin) 居然有 KVM 支持. 于是决定不再纸上谈兵, 开始了实战之旅, 尝试在我的手机上跑跑 KVM 虚拟机(我的手机又遭殃了).  本篇博客可能较长, 应该会包含以下内容:  目前安卓手机上 KVM 的现况 ARM SoC 启动流程以及特权等级 我是如何为我的 chopin 移植 OSS

https://blog.lyc8503.site/post/android-kvm-on-mediatek/

**Comment by xuxin1955** [comment]
为什么核心开满8核反而更卡了？

**Reply by lyc8503** [reply]
因为有大小核，调度可能有问题，而且核心开多了，Windows 启动时识别也可能有问题，你看任务管理器识别到几个逻辑处理器？可能最后就识别到一核。

**Comment by rbqvq** [comment]
twrp只适配了a11好像 刷了a12底包怎么刷呢

**Reply by lyc8503** [reply]
我就留着官方 recovery (约等于没有), 没用 TWRP, 系统本身是 fastboot 线刷的, 直接 `fastboot flash system xxx.img` 即可.

**Comment by Selflocking** [comment]
太酷了

**Comment by gitercn** [comment]
preloader.bin就是类似电脑上的BIOS吧，换未熔丝的SoC硬改解EL2解锁BL也需要修改这个文件吗？现在小米手机几乎解不了BL锁了，估计将来只能换公版SoC了。

**Reply by lyc8503** [reply]
其实可以买个能解锁的机器当钉子户，比如旧小米或一加或海外机器，现代手机的处理器性能已经过剩到不知道哪里去了。

**Comment by seriaTvT** [comment]
太酷了，去年看到文章之后就一直想复现，奈何时间不允许。最近不死心我又来试一试了hhh
我 fork 了你的仓库，Action 有些改动，指定 ubuntu-22.04 和 upload-artifact@v4 之后可以正常编译。
不过我尝试刷入这个内核没有成功，第一屏无限重启。我的操作是先用 Android Image Kitchen 拆原版 boot.img，把 Action 产物中 Image.gz 和 dts/mediatek/chopin.dtb 重命名 boot.img-kernel 和 boot.img-dtb 替换重新打包，然后 fastboot flash boot 刷进去。
是不是我的理解哪里有问题呀？我是小白，没有太多的刷机经验ww

**Reply by lyc8503** [reply]
年代久远，具体的细节有些记不清了

你应该拆原版 Android 12 的 boot.img 替换内核；然后你用的这个 Android Image Kitchen 看起来有点旧了，不知道对新版安卓支持的是否好，可以多 Google 几个新的 boot.img 拆包/打包工具试试，当时我用的是一个网页打包工具，现在不见了qvq

或者我这里有一些 Note 10 Pro 打包好的 boot.img，你也可以试试：
https://pan.lyc8503.net/Public/%E5%9B%BA%E4%BB%B6/MiNote10Pro/2024_KVM%E7%9B%B8%E5%85%B3/%E9%87%8D%E8%A6%81!_%E5%8C%85%E5%90%AB%E6%89%93%E5%8C%85%E7%9A%84%E5%A4%9A%E4%B8%AA%20kernel_chopin_global_images_V13.0.7.0.SKPMIXM_20221110.0000.00_12.0_global_ee0c05a15b/images

**Reply by seriaTvT** [reply]
谢谢你！因为看到 preloader.bin 的问题，所以我就干脆直接先刷了miui12.5，然后用这个12.5的 boot.img 改的
刚刚换了 magiskboot 打包发现也不行！我理解上编译出来的这个 image.gz 就是 kernal，chopin.dtb 就是 dtb，应该没错吧？ 能不能讲讲你是怎么打包的呀谢谢 ovo
又及、试了试你的 boot.img 也无限重启了，果然是因为底包不一样嘛？要不我试试先刷miui13然后再刷 MIUI 12 的 preloader.bin......

**Reply by lyc8503** [reply]
忘记提了，你只需要用到 Actions 里编译出来的 Image，得到的 dtb 文件应该无法使用，需要继续用原厂解包的 dtb。

> 要不我试试先刷miui13然后再刷 MIUI 12 的 preloader.bin

如果你用的是 android S 那个仓库，你必须使用 MIUI 13，否则无法启动。

**Reply by seriaTvT** [reply]
感谢！已经成功啦~ 是我最开始没仔细想就直接刷进 MIUI12 了。现在我刷到了 MIUI13，然后刷了 preloader.bin，再刷了没替换 dtb 打包的内核就OK了。我也不是很喜欢MIUI，后续也刷个新版本GSI试试。非常感谢你的文章和源码，刷刷刷很有趣ww
![Screenshot demonstrating Fedora Linux running successfully inside a KVM virtual machine on a MediaTek MTK1100 device, evidenced by a fastfetch terminal output alongside a related blog post.](https://pediastrum.com/wp-content/uploads/2026/02/scmtk1100.webp)

**Comment by mylifewaiting** [comment]
hi，请教下，我用pixel 8装了win11 arm系统，网卡驱动安装的时候就崩溃重启了，必现，要怎么搞，
